sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -1243,13 +1243,14 @@ def test_gradient_boosting_with_init_pipeline():
 
     X, y = make_regression(random_state=0)
     init = make_pipeline(LinearRegression())
-    gb = GradientBoostingRegressor(init=init)
+    gb = GradientBoostingRegressor(init=init).request_sample_weight(fit=True)
```

Comment:
Does this test fail if one does not call `request_sample_weight`?

Reply:
Not in the current implementation but this is an intended feature, yes. It was a late tweak to the approach.

Reply:
Sorry, I missed the context. This test will work with or without calling `request_sample_weight`. I don't think this needed changing here.